<?xml version="1.0"?>
<!--

-->
<launch>
<include file="$(find mocap_qualisys)/resources/camera_comp/transforms.launch" />
<group>
    <param name="zed_description" textfile="$(find mocap_qualisys)/resources/camera_comp/zedm.urdf" />
    <node name="zed_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">  
        <remap from="robot_description" to="zed_description" />
    </node>
</group>

<group ns="zed">
  <node pkg="mocap_qualisys"
        type="synchronize_mocap.py"
        name="sync_mocap_zed"
        output="screen">
    <param name="parent_topic" value="/qualisys/svea3/pose"/>
    <param name="child_topic" value="zed_node/odom"/>
    <param name="child_frame" value="base_link"/>
    <param name="sync_topic" value="/comp/sync"/>
    <param name="sync_at_init" value="False"/>
  </node>
</group>

<group ns="rs">
  <node pkg="mocap_qualisys"
         type="synchronize_mocap.py"
         name="sync_mocap_rs"
         output="screen">
    <param name="parent_topic" value="/qualisys/svea3/pose"/>
    <param name="child_topic" value="realsense_camera/odom/sample"/>
    <param name="child_frame" value="rs_base_link"/>
    <param name="sync_topic" value="/comp/sync"/>
    <param name="sync_at_init" value="False"/>
  </node>
</group>   

<node 
  pkg="rosbag"
  type="play"
  name="rosbag_player"
  output="screen"
  args="--clock $(find mocap_qualisys)/resources/camera_comp/camera_comparision.bag"/>

</launch>
